packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -178,38 +220,42 @@ function useProcessMedia({
    * @param {string} end Time stamp of end time of new video. Example '00:02:00'.
    */
   const trimExistingVideo = useCallback(
-    ({ resource: oldResource, canvasResourceId, start, end }) => {
+    ({ resource: oldResource, elementId, start, end }) => {
       const { id: resourceId, ...oldResourceWithoutId } = oldResource;
       const { src: url, mimeType, poster, isMuted, isOptimized } = oldResource;
 
       const trimData = {
         original: resourceId,
         start,
         end,
+        elementId,
       };
 
       const onUploadStart = () =>
-        updateExistingElements(canvasResourceId, {
+        updateExistingElement(elementId, {
           trimData,
         });
 
       const onUploadError = () =>
-        updateExistingElements(canvasResourceId, {
+        updateExistingElement(elementId, {
           trimData: oldResource.trimData || {},
         });
 
       const onUploadSuccess = ({ resource }) => {
         const oldCanvasResource = {
           alt: oldResource.alt,
-          id: canvasResourceId,
+          id: elementId,
         };
-        copyResourceData({ oldResource: oldCanvasResource, resource });
+        copyResourceDataByElementId({
+          oldResource: oldCanvasResource,
+          resource,
+        });
         postProcessingResource(resource);
       };
 
       const onUploadProgress = ({ resource }) => {
-        const newResourceWithCanvasId = { ...resource, id: canvasResourceId };
-        updateExistingElements(resourceId, {
+        const newResourceWithCanvasId = { ...resource, id: elementId };
```

Comment:
This would override the resource ID with the element ID. We don't want that. Instead, we want to preserve the original resource ID.

Reply:
Noting: We have a reference to the `original` resourceId under `original`

```
const trimData = {
        original: resourceId,
        start,
        end,
        elementId, // I might be able to remove this completely
      };
      ```

Reply:
Why are we using the element id in place of a resource id here. This is really confusing to me.